sdk/tests/DeserializationTests.cc

```diff
@@ -0,0 +1,254 @@
+#include <proto/transaction_response.pb.h>
+#include <proto/crypto_get_account_balance.pb.h>
+
+using namespace Hedera;
+
+class DeserializationTests : public ::testing::Test
+{
+protected:
+  [[nodiscard]] inline const uint64_t getTestShardNum() const { return mShardNum; }
+  [[nodiscard]] inline const uint64_t getTestRealmNum() const { return mRealmNum; }
+  [[nodiscard]] inline const uint64_t getTestContractNum() const { return mContractNum; }
+  [[nodiscard]] inline const int32_t  getTestCents() const { return cents; }
+  [[nodiscard]] inline const int32_t  getTestHbar() const { return hbar; }
+  [[nodiscard]] inline const uint64_t getTestSeconds() const { return seconds; }
+  [[nodiscard]] inline const Hbar     getTestBalance() const { return mBalance; }
+  [[nodiscard]] inline const AccountId& getTestAccountId() const { return mAccountId; }
+  [[nodiscard]] inline const AccountId& getTestAccountIdFrom() const { return mAccountIdFrom; }
+  [[nodiscard]] inline const AccountId& getTestAccountIdTo() const { return mAccountIdTo; }
+  [[nodiscard]] inline const std::shared_ptr<ECDSAPublicKey>& getTestECDSAPublicKey() const { return mPublicKeyFromString; }
+
+private:
+  const uint64_t mShardNum = 1;
+  const uint64_t mRealmNum = 2;
+  const uint64_t mContractNum = 3;
+  const int32_t cents = 2;
+  const int32_t hbar = 1;
+  const uint64_t seconds = 100ULL;
+  const Hbar mBalance = Hbar(100LL);
+  const AccountId mAccountId = AccountId(0ULL, 0ULL, 10ULL);
+  const AccountId mAccountIdFrom = AccountId(4ULL);
+  const AccountId mAccountIdTo = AccountId(3ULL);
+  const std::unique_ptr<ECDSAPrivateKey> mPrivateKey = ECDSAPrivateKey::generatePrivateKey();
+  const std::shared_ptr<PublicKey> mPublicKeyFromPrivate = mPrivateKey->getPublicKey();
+  const std::shared_ptr<ECDSAPublicKey> mPublicKeyFromString = ECDSAPublicKey::fromString(mPublicKeyFromPrivate->toString());
+};
+
+TEST_F(DeserializationTests, DeserializeExchangeRateFromProtobufTest)
+{
+  // Given
+  const int32_t testCents = getTestCents();
+  const int32_t testHbar = getTestHbar();
+  const uint64_t testSeconds = getTestSeconds();
+
+  auto testProtoExchangeRate = std::make_unique<proto::ExchangeRate>();
+  proto::TimestampSeconds *testProtoExchangeRateSecs = testProtoExchangeRate->mutable_expirationtime();
+
+  testProtoExchangeRate->set_centequiv(testCents);
+  testProtoExchangeRate->set_hbarequiv(testHbar);
+  testProtoExchangeRateSecs->set_seconds(testSeconds);
+
+  // When
+  ExchangeRate exchangeRate = ExchangeRate::fromProtobuf(*testProtoExchangeRate);
+  
+  // Then
+  EXPECT_EQ(exchangeRate.getCurrentExchangeRate(), testCents / testHbar);
+  EXPECT_TRUE(exchangeRate.getExpirationTime().has_value());
+  EXPECT_EQ(exchangeRate.getExpirationTime().value().time_since_epoch().count(),
+            internal::TimestampConverter::fromProtobuf(*testProtoExchangeRateSecs).time_since_epoch().count());
+}
+
+TEST_F(DeserializationTests, DeserializeExchangeRateSetFromProtobufTest)
```

Comment:
To write this test I used test [ProtobufExchangeRate](https://github.com/hashgraph/hedera-sdk-cpp/blob/c54139e8bc8f507e1f08d052b15eec3d188a0bd0/sdk/tests/ExchangeRateTest.cc#L43) from [ExchangeRateTest.cc](https://github.com/hashgraph/hedera-sdk-cpp/blob/main/sdk/tests/ExchangeRateTest.cc).

Reply:
Test `DeserializeExchangeRateSetFromProtobufTest ` moved into `ExchangeRateTest.cc` in commit https://github.com/hashgraph/hedera-sdk-cpp/pull/176/commits/a3879f92d8fa2ce3ce92699b7c572a84fbaca461.